library/dnet.c

```diff
@@ -474,13 +866,15 @@ static int dnet_iterator_callback_common(void *priv, struct dnet_raw_id *key, ui
 	response->flags = flags;
 	dnet_convert_iterator_response(response);
 
-	/* Data */
-	err = dnet_read_ll(fd, (char *)position, dsize, data_offset);
-	if (err)
-		goto err_out_exit;
+	if (dsize) {
```

Comment:
The real data corruption bug was introduced because this check was removed.
Originally code was safe and then *optimized* to hide this check, this optimization removed correct offset calculation, which ended up with overwriting iterator response with data. Apparently you do not use this code at all.

I fixed this bug and returned back explicit check.